design/bgp-bfd.md

```diff
@@ -0,0 +1,91 @@
+# BFD enablement in BGP
+
+## Summary
+
+The purpose of this enhancement is to use enable [BFD](https://en.wikipedia.org/wiki/Bidirectional_Forwarding_Detection) to be used
+in conjunction with BGP.
+
+## Motivation
+
+BFD provides a quicker path failure detection than BGP, and using them together will allow users to provide a better
+service.
+
+BFD for BGP is supported by FRR [out of the box](http://docs.frrouting.org/en/latest/bfd.html#bgp-bfd-configuration).
+
+
+### Goals
+
+* Provide a way to establish BFD on BGP session.
+* Provide a way to configure the parameters of the BFD session associated with a BGP session.
```

Comment:
I think there are a lot of tunables for this? Do you have any suggestion what would be the minimum we would need to expose? I think MetalLB has a reputation as simple to configure and "just works out of the box" so we should really try to minimize this as much as possible?

Reply:
Should this design include references to the operator and any work that needs to be done there?

Reply:
> I think there are a lot of tunables for this? Do you have any suggestion what would be the minimum we would need to expose? I think MetalLB has a reputation as simple to configure and "just works out of the box" so we should really try to minimize this as much as possible?

It is in the design section below. There are 3 properties that we can set for each session. I guess we can define defaults or leverage and declare the frr defaults if we don't specify them.

Reply:
> Should this design include references to the operator and any work that needs to be done there?

I will add a section, makes total sense